svsim/src/main/scala/Workspace.scala

```diff
@@ -0,0 +1,389 @@
+package svsim
+
+import java.io.{BufferedReader, BufferedWriter, File, FileWriter, InputStreamReader}
+import java.nio.file.Paths
+import java.lang.ProcessBuilder.Redirect
+
+case class ModuleInfo(
+  name:  String,
+  ports: Seq[ModuleInfo.Port]) {
+  private[svsim] val instanceName = "dut"
+}
+object ModuleInfo {
+  case class Port(
+    name:       String,
+    isSettable: Boolean = false,
+    isGettable: Boolean = false) {
+    assert(name.matches("^[a-zA-Z0-9\\-_]*$"))
+  }
+}
+
+object Workspace {
+  val testbenchModuleName: String = "svsimTestbench"
+}
+final class Workspace(
```

Comment:
I think this is possibly not a good idea to declare `Workspace` Here. These is kind of pitfall that we should learn from: `rocket-chip` did it in its tests generation, chiseltest force to use `test_run_dir`, and use Scalatest of cause, I think decoupling function and build system will be neat.